src/Xrefcheck/Scan.hs

```diff
@@ -193,83 +192,101 @@ readDirectoryWith mode config scanner root = do
       RdmBothTrackedAndUtracked -> liftA2 (<>) getTrackedFiles getUntrackedFiles
 
     getTrackedFiles = readCreateProcess
-      (shell "git ls-files"){cwd = Just $ unCanonicalPath root} ""
+      (shell "git ls-files -z"){cwd = Just root} ""
     getUntrackedFiles = readCreateProcess
-      (shell "git ls-files --others --exclude-standard"){cwd = Just $ unCanonicalPath root} ""
+      (shell "git ls-files -z --others --exclude-standard"){cwd = Just root} ""
 
-    scanFile :: CanonicalPath -> IO (CanonicalPath, a)
+    fileLines :: String -> [String]
+    fileLines = go DL.empty
```

Comment:
This is `lines` but with null line delimiters. Do we have already a function available to split a list in this way? We could also use the [split](https://hackage.haskell.org/package/split) package.

Reply:
I also don't know about such a function, that's sad that it seems to be absent in base.

Yeah, `split` library is an option, but I think let's not add extra dependencies for just one use case. I believe via `Data.List.break` this can be implemented relatively short.